kernel/fatal.c

```diff
@@ -0,0 +1,92 @@
+#include <kernel.h>
+#include <kernel_internal.h>
+#include <misc/printk.h>
+#include <arch/cpu.h>
+#include <logging/log_ctrl.h>
+#include <fatal.h>
+
+/* LCOV_EXCL_START */
+FUNC_NORETURN __weak void z_arch_system_halt(unsigned int reason)
+{
+	ARG_UNUSED(reason);
+
+	for (;;) {
+		k_cpu_idle();
+	}
+}
+/* LCOV_EXCL_STOP */
+
+/* LCOV_EXCL_START */
+__weak void k_sys_fatal_error_handler(unsigned int reason,
+				      const NANO_ESF *esf)
+{
+	ARG_UNUSED(esf);
+
+	printk("Halting system.\n");
+	(void)irq_lock();
+
+	z_arch_system_halt(reason);
+}
+/* LCOV_EXCL_STOP */
+
+void z_fatal_error(unsigned int reason, const NANO_ESF *esf)
+{
+#ifdef CONFIG_THREAD_NAME
+	const char *thread_name = k_thread_name_get(k_current_get());
+#endif
+	LOG_PANIC();
+
+#ifdef CONFIG_PRINTK
+	printk("******* ");
+
+	switch (reason) {
+	case K_ERR_CPU_EXCEPTION:
+		printk("CPU Exception\n");
+		break;
+
+	case K_ERR_SPURIOUS_IRQ: {
+		printk("Unhandled interrupt\n");
+		break;
+	}
+#if defined(CONFIG_STACK_CANARIES) || defined(CONFIG_STACK_SENTINEL) || \
+		defined(CONFIG_HW_STACK_PROTECTION) || \
+		defined(CONFIG_USERSPACE)
```

Comment:
FWIW: for clarity, maybe define a "HAVE_ERR_STACK_CHK_FAIL" elsewhere?  Or just have a case for this (like 3-4 instructions and the static string) defined always, even on configurations where we can't actually emit such an error.

Reply:
yeah more trouble than it's worth, I just removed the ifdef.